### PR TITLE
Add DTrace

### PR DESCRIPTION
**Pull request by xen0l**
**https://github.com/xen0l/awesome-dtrace / http://awesome-dtrace.com**

DTrace is a dynamic tracing framework, which allows safe observation of productions systems. Over the time, many programming languages, operating systems and frameworks added support for various DTrace probes and made problem troubleshooting available via DTrace. DTrace is a wonderful tool and more people should know about it and use it.

# ALL THE BELOW CHECKBOXES SHOULD BE DONE AND CHECKED AND THE ABOVE PLACEHOLDERS SHOULD BE FILLED IN BEFORE SUBMITTING A PULL REQUEST!

- [x] I have read and understood the [contribution guidelines](https://github.com/sindresorhus/awesome/blob/master/contributing.md) and the [instructions for creating a list](https://github.com/sindresorhus/awesome/blob/master/create-list.md).
- [x] This pull request has a descriptive title. *(For example: `Add Node.js`)*
- The list I added
	- [x] **has been around for at least 20 days,**
	- [x] is not a duplicate,
	- [x] only has awesome items with descriptions *(Awesome lists are curations of the best, not of everything)*,
	- [x] contains the [awesome badge](https://github.com/sindresorhus/awesome/blob/master/awesome.md#awesome-badge) on the right side of the list heading,
	- [x] has a `Table of Contents` as the first section,
	- [x] has an [appropriate license](https://github.com/sindresorhus/awesome/blob/master/awesome.md#choose-an-appropriate-license) *(Meaning something like CC0, NOT a code licence like MIT, BSD, Apache, etc)*,
	- [x] has [contribution guidelines](https://github.com/sindresorhus/awesome/blob/master/awesome.md#include-contribution-guidelines),
	- [x] has been checked for inconsistent formatting, spelling, and grammar *(This includes starting each link description with an uppercase character and ending it with a dot, and consistent naming)*,
	- [x] has been added at the bottom of the appropriate category.

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -431,7 +431,7 @@ Check out my [blog](https://blog.sindresorhus.com) and follow me on [Twitter](ht
 - [Colorful](https://github.com/Siddharth11/Colorful) - Resources to choose your next color scheme.
 - [Steam](https://github.com/scholtzm/awesome-steam) - Digital distribution platform.
 - [Bots](https://github.com/hackerkid/bots) - Building bots.
-
```
Comment (sindresorhus) [inline readme.md]: Don't remove newline.

File: readme.md [hunk 2]
```diff
@@ -431,7 +431,7 @@ Check out my [blog](https://blog.sindresorhus.com) and follow me on [Twitter](ht
...
+- [DTrace](https://github.com/xen0l/awesome-dtrace) - DTrace - dynamic tracing framework.
```
Comment (sindresorhus) [inline readme.md]: `DTrace - dynamic tracing framework.` => `Dynamic tracing framework.`

**Comment by sindresorhus** [comment]
> ALL THE BELOW CHECKBOXES SHOULD BE DONE AND CHECKED AND THE ABOVE PLACEHOLDERS SHOULD BE FILLED IN BEFORE SUBMITTING A PULL REQUEST!

All the checkboxes are not checked.

**Comment by xen0l** [comment]
Yes, I know and am working on it. I have to recreate ToC and recheck everything. It should be finished soon.

**Comment by xen0l** [comment]
I have checked all the checkboxes now.

**Comment by sindresorhus** [comment]
>  has been checked for inconsistent formatting, spelling, and grammar (This includes starting each link description with an uppercase character and ending it with a dot, and consistent naming),

Not done completely. Note "This includes starting each link description with an uppercase character and ending it with a dot".

-

Some nitpick:

- `node.js` => `Node.js`
- https://github.com/xen0l/awesome-dtrace/blob/master/LICENSE should include the actual license
- Drop `Awesome DTrace` in the table of contents (TOC) and give the TOC a top-level heading.
- I don't really see the point in mentioning this: `Layout inspired by awesome-python.`
- `Learn DTrace` => `Learn`
- Align the logo to the right ``.


The TOC is more verbose than useful. It's only meant to be a way to quickly scan and go to the main sections. Here's how I think it should look instead:

```
## Contents

- [Learn](#learn-dtrace)
- [Articles](#articles)
- [Videos](#videos)
- [Software](#software)
- [Tools](#tools)
- [Community](#community)
- [Contributing](#contributing)
```

**Comment by sindresorhus** [comment]
ping

**Comment by xen0l** [comment]
I'm on holiday this week and might be able to fix it during the weekend. Will try sooner.

**Comment by 78f3bcb11** [comment]
Asombroso

**Comment by xen0l** [comment]
I fixed everything requested and rebased pull request. Hopefully, it's everything.

File: readme.md [hunk 3]
```diff
@@ -436,7 +436,7 @@ Check out my [blog](https://blog.sindresorhus.com) and follow me on [Twitter](ht
 - [Bots](https://github.com/hackerkid/bots) - Building bots.
 - [Site Reliability Engineering](https://github.com/dastergon/awesome-sre)
 - [Empathy in Engineering](https://github.com/KimberlyMunoz/empathy-in-engineering) - Building and promoting more compassionate engineering cultures.
-
```
Comment (sindresorhus) [inline readme.md]: Don't remove the newline

File: readme.md [hunk 4]
```diff
@@ -436,7 +436,7 @@ Check out my [blog](https://blog.sindresorhus.com) and follow me on [Twitter](ht
...
+- [DTrace](https://github.com/xen0l/awesome-dtrace) - DTrace - Dynamic tracing framework.
```
Comment (sindresorhus) [inline readme.md]: I've already commented about this... => https://github.com/sindresorhus/awesome/pull/701#discussion_r69030389

**Comment by sindresorhus** [comment]
> node.js => Node.js

Not fixed.

> Add the section title to Table of Contents.

In the contributing.md it should be `Contents`.

**Comment by xen0l** [comment]
Updated PR. I also replaced ToC with Contents and Node.js as you requested.

**Comment by xen0l** [comment]
I also added awesome-lint integration via Travis and made the list awesome-lint clean.

**Comment by sindresorhus** [comment]
https://github.com/sindresorhus/awesome/pull/701/commits/d8b69d0e59b21455c00510524de90195abf26cdd is incorrect. No idea why you're adding a `-` there. https://github.com/sindresorhus/awesome/pull/701#discussion_r72891914 is also not fixed.

**Comment by sindresorhus** [comment]
I appreciate you trying, but this has taken too long. I'll just fix the remaining issues when merging ;)

**Comment by xen0l** [comment]
We might have misunderstood each others. Thanks for your patience! :)

**Pull request merged**